src/mutations/article/updateTagSetting.ts

```diff
@@ -54,20 +61,86 @@ const resolver: MutationToUpdateTagSettingResolver = async (
 
       params = { owner: viewer.id, editors: _uniq([...tag.editors, viewer.id]) }
       break
-    case GQLUpdateTagSettingType.leave:
+    }
+    case GQLUpdateTagSettingType.leave: {
       // if tag has no owner or owner is not viewer, throw error
       if (!tag.owner || (tag.owner && tag.owner !== viewer.id)) {
         throw new ForbiddenError('viewer has no permission')
       }
 
-      const isMatty = viewer.id === environment.mattyId
-      const editors = isMatty
+      // remove viewer from editors
+      const newEditors = isMatty
         ? undefined
         : (tag.editors || []).filter((item: string) => item !== viewer.id)
-      params = { owner: null, editors }
+      params = { owner: null, editors: newEditors }
+      break
+    }
+    case GQLUpdateTagSettingType.add_editor: {
+      // only viewer can add editors
+      if (!isOwner) {
+        throw new ForbiddenError('viewer has no permission')
+      }
+      if (!editors || editors.length === 0) {
+        throw new UserInputError('editors are invalid')
+      }
+
+      // gather valid editors
+      const newEditors = (
+        await Promise.all(
+          editors.map(async (editor) => {
+            const { id: editorId } = fromGlobalId(editor)
+            const user = await userService.baseFindById(editorId)
```

Comment:
seems no need for fresh data, should we use `.dataloader.load` or `.dataloader.loadMany` instead?

Reply:
Or, since the update operation will be failed if ids are invalid, do we really need this DB checking?

Reply:
Nice idea. Removed checks. Thanks.